features/appmgt/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/apis/v1/lifecycle_api_router.jag

```diff
@@ -131,364 +131,405 @@ require('/modules/publisher.js').exec(function (ctx) {
+            log.error("Insufficient permissions to perform '" + action + "' on " + appType);
+            response.status = 401;
+            response.contentType = "application/json";
+            content.status = "Access Denied";
+            content.messages.push("You do not have publisher permissions to perform '"
+                                  + action + "' on " + appType + "s");
+            response.content = stringify(content);
+            return;
+        }
+
+        var carbon = require('carbon');
+        var artifactManager = rxtManager.getSystemArtifactManager(appType, session);
+
+        try {
+            var currentArtifact = artifactManager.get(uuid);
+        } catch (e) {
+            // this exception is thrown by
+            // GenericArtifactManager.getGenericArtifact(id) throws GovernanceException
+            log.error("The asset with UUID " + uuid + " could not be retrieved.");
+            log.error(e);
+            response.status = 404;
+            response.contentType = "application/json";
+            content.status = "Not Found";
+            content.messages.push("The " + appType + " with UUID " + uuid + " cannot be retrieved");
+            response.content = stringify(content);
+            return;
+        }
+
+        // grant permissions
+        var roles = (context.content) ? context.content.roles : null;
+        if (roles && (roles.length > 0)) {
+            var currentRole = null;
+            try {
+                roles.forEach(function (role) {
+                    currentRole = role;
+                    userManager.authorizeRole(currentRole, currentArtifact.path,
+                                              carbon.registry.actions.GET);
+                });
+            } catch (e) {
+                log.error("Cannot grant permissions to role '" + currentRole
+                          + "' to perform action '" + carbon.registry.actions.GET + "'");
+                log.error(e);
+                // continue work flow
+            }
+        }
+
+        // check permissions to perform life cycle action
+        try {
+            if (!permissions.isLCActionsPermitted(user.username, currentArtifact.path,
+                                                  userManager)) {
+                log.error("Insufficient permissions to perform '" + action + "' on " + appType
+                          + " for user " + user.username);
+                response.status = 401;
+                response.contentType = "application/json";
+                content.status = "Insufficient Permissions";
+                content.messages.push("You do not have permissions to perform action '"
+                                      + action + "' on version "
+                                      + currentArtifact.attributes.overview_version
+                                      + " of '" + currentArtifact.attributes.overview_name + "' "
+                                      + appType);
```

Comment:
Just a suggestion...
You do not have permissions to perform 'publish' action on webapp. app name: application01 - app version: 1.0.0
vs
You do not have permissions to perform action 'publish' on version 1.0.0 of 'application01' webapp